hack/manifest-gen/README.md

```diff
@@ -0,0 +1,44 @@
+ECK Manifest Generator
+======================
+
+Utility for generating ECK installation manifests.
+
+Quickstart
+----------
+
+NOTE: Requires Docker and Make to be installed on your computer.
+
+Cluster-wide installation with access to all namespaces:
+
+```shell
+make docker-gen-global | kubectl apply -f -
+```
+
+Installation restricted to a single namespace named "myns":
+
+```shell
+make docker-gen-restricted RESTRICTED_NS=myns | kubectl apply -f -
+```
```

Comment:
It is more likely that users will have Docker installed on their machines rather than Go. Asking them to install Go and compile a binary is not a great user experience. That is one of the reasons why I was keen on the distribution model enabled by my initial proposal.

With the code staying inside the cloud-on-k8s repo,  the options as I see them are:
1. Tell users to build the manifest-gen container image locally
2. Make the manifest-gen container image available in the public registry
3. Bundle the manifest-gen binary with the ECK container image

Reply:
I think I'd be fine with releasing the manifest-gen along with the ECK image, especially if they are versioned together (your option 3)?

However I think we should also consider:

4. Release the manifest-gen as a standalone binary (windows/osx/linux) in our github releases

I imagine most users would either run it from the ECK Docker image, or from the standalone binary.
I would consider that users who run things from the source code after a git clone can manage installing Go?

I'm fine keeping things as they are if you think it makes more sense for now, and revisit once we settle on the release/distribution mechanism.

Reply:
+1 to discuss an other release mechanism later (`curl`, `brew install elastic/tap/eckctl`, `yum install eckctl` ...)